pkg/client/client.go

```diff
@@ -1245,6 +1244,52 @@ func (c *Client) WaitForSecretByNsName(ctx context.Context, obj types.Namespaced
 	return c.WaitForSecret(ctx, &secret)
 }
 
+func (c *Client) WaitForConfigMap(ctx context.Context, cm *v1.ConfigMap) (*v1.ConfigMap, error) {
+	var result *v1.ConfigMap
+	var lastErr error
+	if err := wait.Poll(1*time.Second, 5*time.Minute, func() (bool, error) {
+		var err error
+
+		result, err = c.kclient.CoreV1().ConfigMaps(cm.Namespace).Get(ctx, cm.Name, metav1.GetOptions{})
+		if err != nil {
+			lastErr = err
+			return false, nil
+		}
+
+		if len(result.Data) == 0 {
+			lastErr = errors.New("configmap contains no data")
```

Comment:
nit: either the error message should say `configmap` or `ConfigMap`, I'll let you decide, I suggested this since online 1276 you had written `ConfigMap`. Also shouldn't we say which ConfigMap the error refers to?
```suggestion
			lastErr = errors.New("ConfigMap contains no data")
```

Reply:
I used the lower-case version here because the convention for error messages is that they shouldn't be capitalized.